libdill.c

```diff
@@ -28,21 +28,45 @@
 #include <mach/mach_time.h>
 #endif
 
+#if defined(__x86_64__) || defined(__i386__)
+#include <x86intrin.h>
+#endif
+
 #include "cr.h"
 #include "libdill.h"
 #include "pollset.h"
 #include "utils.h"
 
-int64_t now(void) {
+static __attribute__((noinline)) uint64_t dill_now_measure(void) {
+    uint64_t start_rdtsc = __rdtsc();
+    usleep(100);
+    uint64_t stop_rdtsc = __rdtsc();
+    int64_t diff_rdtsc = stop_rdtsc - start_rdtsc;
+    if(diff_rdtsc < 0) diff_rdtsc = -diff_rdtsc;
```

Comment:
What's the point? If the thread was migrated to different core during the measurement the number is going to be completely off.

Also, power saving measures on laptops can affect clock speed. So the number you measure here may not be correct later on. I would just go with a fixed number.